configure.py

```diff
@@ -1794,6 +1799,12 @@ def configure_abseil(build_dir, mode, mode_config):
             command = reloc/build_deb.sh --reloc-pkg $in --builddir $out
         rule unified
             command = unified/build_unified.sh --mode $mode --unified-pkg $out
+        rule rust_header
+            command = cxxbridge $in > $out
+            description = RUST_HEADER $out
+        rule rust_lib
+            command = CARGO_HOME=build/rust/.cargo cargo build --release --manifest-path=rust/Cargo.toml --target-dir=build/rust -p ${{pkg}}
```

Comment:
This creates the `.a` library in the `build/rust/release/` directory, but other rules expect it to be in the `build/{mode}/rust/release/` directory and the build fails for me because of that.

Reply:
You are building the rust libraries in release profile regardless of the actual scylla build mode. It's not great because most of the time (in dev mode) we don't want a fully optimized build and would prefer the compilation time to be short instead. Preferably, we would like to customize the compiler options for each build mode.

I suggest looking into [cargo profiles](https://doc.rust-lang.org/cargo/reference/profiles.html). The dev and release profiles are already defined by default (but might need some tweaking), so you need to define custom debug, sanitize and coverage profiles. The `rust_lib` could select the appropriate profile with `--profile {mode}`.

As you are using a cargo workspace, note that the profiles need to be defined in workspace's `Cargo.toml`, not the crates' `Cargo.toml`.

Reply:
> This creates the `.a` library in the `build/rust/release/` directory, but other rules expect it to be in the `build/{mode}/rust/release/` directory and the build fails for me because of that.

I missed this when fixing another issue, will update soon

> I suggest looking into [cargo profiles](https://doc.rust-lang.org/cargo/reference/profiles.html). The dev and release profiles are already defined by default (but might need some tweaking), so you need to define custom debug, sanitize and coverage profiles. The `rust_lib` could select the appropriate profile with `--profile {mode}`.

The issue I had with cargo profiles was that I couldn't add the `debug` profile, and the `dev` profile was building files in the `debug` directory. I postponed fixing this problem because there was practically zero rust files to compile for now so it doesn't make a noticeable difference yet, but I was planning to add the profiles alongside the first rust contribution - using rust wasmtime instead of the c++ bindings